python/az/aro/azext_aro/tests/latest/unit/test_validators.py

```diff
@@ -0,0 +1,419 @@
+        "subscription",
+        None,
+        Mock(cli_ctx=None),
+        InvalidArgumentValueError
+    ),
+    (
+        "should raise InvalidArgumentValueError exception when subnet has child namespace",
+        Mock(key='192.168.0.0/28', vnet=False),
+        'key',
+        True,
+        {"subscription": "subscription", "namespace": "MICROSOFT.NETWORK", "type": "virtualnetworks", "last_child_num": 1, "child_namespace_1": "something"},
+        "subscription",
+        None,
+        Mock(cli_ctx=None),
+        InvalidArgumentValueError
+    ),
+    (
+        "should raise InvalidArgumentValueError exception when child type subnet do not equal subnets",
+        Mock(key='192.168.0.0/28', vnet=False),
+        'key',
+        True,
+        {"subscription": "subscription", "namespace": "MICROSOFT.NETWORK", "type": "virtualnetworks", "last_child_num": 1, "child_type_1": "this_should_be_subnets"},
+        "subscription",
+        None,
+        Mock(cli_ctx=None),
+        InvalidArgumentValueError
+    ),
+    (
+        "should raise InvalidArgumentValueError exception when client.subnets.get raises CLIInternalError because client.subnets.get() raises Exception exception",
+        Mock(key='192.168.0.0/28', vnet=False),
+        'key',
+        True,
+        {"subscription": "subscription", "namespace": "MICROSOFT.NETWORK", "type": "virtualnetworks", "last_child_num": 1, "child_type_1": "subnets"},
+        "subscription",
+        Mock(**{"subnets.get.side_effect": Exception}),
+        Mock(cli_ctx=None),
+        CLIInternalError
+    ),
+    (
+        "should raise InvalidArgumentValueError exception when client.subnets.get() raises ResourceNotFoundError exception",
+        Mock(key='192.168.0.0/28', vnet=False),
+        'key',
+        True,
+        {"subscription": "subscription", "namespace": "MICROSOFT.NETWORK", "type": "virtualnetworks", "last_child_num": 1, "child_type_1": "subnets", "resource_group": None, "name": None, "child_name_1": None},
+        "subscription",
+        Mock(**{"subnets.get.side_effect": ResourceNotFoundError("")}),
+        Mock(cli_ctx=None),
+        InvalidArgumentValueError
+    ),
+    (
+        "should not raise any exception",
+        Mock(key='192.168.0.0/28', vnet=False),
+        'key',
+        True,
+        {"subscription": "subscription", "namespace": "MICROSOFT.NETWORK", "type": "virtualnetworks", "last_child_num": 1, "child_type_1": "subnets", "resource_group": None, "name": None, "child_name_1": None},
+        "subscription",
+        Mock(**{"subnets.get.return_value": None}),
+        Mock(cli_ctx=None),
+        None
+    )
+]
+
+
+@ pytest.mark.parametrize(
+    "test_description, namespace, key, is_valid_resource_id_mock_return_value, parse_resource_id_mock_return_value, get_subscription_id_mock_return_value, get_mgmt_service_client_mock_return_value, cmd, expected_exception",
```

Comment:
9 parameters - we are likely doing too much.

Reply:
Yes, I agree with that, but the "problem" is that the function under tests calls different MSFT functions and I want this test to be a unit test and don't rely on external calls. For me here is the boundary of the "unit" so I decided to mock those external dependencies. 
In a different context, I would probably not mock them (e,g: if the unit under test was invoking jus functions of the domain layer), but here it makes sense (IMHO).